task/sast-unicode-check/0.1/sast-unicode-check.yaml

```diff
@@ -0,0 +1,265 @@
+      # the cluster will set imagePullPolicy to IfNotPresent
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      volumeMounts:
+      env:
+        - name: KFP_GIT_URL
+          value: $(params.KFP_GIT_URL)
+        - name: FIND_UNICODE_CONTROL_GIT_URL
+          value: $(params.FIND_UNICODE_CONTROL_GIT_URL)
+        - name: PROJECT_NVR
+          value: $(params.PROJECT_NVR)
+        - name: RECORD_EXCLUDED
+          value: $(params.RECORD_EXCLUDED)
+        - name: SOURCE_CODE_DIR
+          value: $(workspaces.workspace.path)
+      script: |
+        #!/usr/bin/env bash
+        set -exuo pipefail
+        . /utils.sh
+        trap 'handle_error $(results.TEST_OUTPUT.path)' EXIT
+
+        SCAN_PROP=""
+
+        # Installation of Red Hat certificates for cloning Red Hat internal repositories
+        curl -sS https://certs.corp.redhat.com/certs/2015-IT-Root-CA.pem > /etc/pki/ca-trust/source/anchors/2015-RH-IT-Root-CA.crt
+        curl -sS https://certs.corp.redhat.com/certs/2022-IT-Root-CA.pem > /etc/pki/ca-trust/source/anchors/2022-IT-Root-CA.pem
+        update-ca-trust
+
+        # Clone the source code from upstream repo
+        clone_unicontrol()
+        {
+            GIT_URL=$(echo ${FIND_UNICODE_CONTROL_GIT_URL} | awk -F'#' '{print $1}')
+            REV=$(echo ${FIND_UNICODE_CONTROL_GIT_URL} | awk -F'#' '{print $2}')
+
+            # Clone find-unicode-control repository
+            git clone --depth=1 "${GIT_URL}" cloned >&2
+            status=$?
+            if [ "${status}" -ne 0 ]; then
+                echo "Error: Failed to clone the repository: ${GIT_URL}" >&2
+                return 1
+            else
+                echo "Message: Succeed to clone the repository: ${GIT_URL}" >&2
+            fi
+
+            if [[ -n "${REV}" ]]; then
+                # Get git url suffix
+                git_url_suffix=$(git -C ./cloned/ rev-parse HEAD)
+                if [[ "${git_url_suffix}" != "${REV}" ]]; then
+                    (cd cloned && git checkout --quiet "${REV}")
+                fi
+                SCAN_PROP="find-unicode-control-git-url:${FIND_UNICODE_CONTROL_GIT_URL}"
+            else
+                SCAN_PROP="find-unicode-control-git-url:${FIND_UNICODE_CONTROL_GIT_URL}#${git_url_suffix}"
+            fi
+
+            # Check usage of find-unicode-control to confirm the tool can work
+            ./cloned/find_unicode_control.py --help >&2
+            status=$?
+            if [ "${status}" -ne 0 ]; then
+                echo "Error: Failed to check usage of find-unicode-control" >&2
+                return 1
+            else
+                echo "Message: Succeed to check usage of find-unicode-control" >&2
+            fi
+        }
+
+        # Find unicode control
+        unicontrol_scan()
+        {
+            LANG=en_US.utf8 ./cloned/find_unicode_control.py -p bidi -v -d -t "${SOURCE_CODE_DIR}/source" \
```

Comment:
According to acceptance criteria of https://issues.redhat.com/browse/OSH-739 a parameter should be provided to override the default arguments passed to the scanner.  Let's introduce the `FIND_UNICODE_CONTROL_ARGS` task parameter for this and `-p bidi -v -d -t` as its default value.

Reply:
Updated